src/evaluation.py

```diff
@@ -352,10 +354,41 @@ def match_answer_for_question(self, current_score, question, marked_answer):
         )
         return delta
 
-    def conditionally_print_explanation(self):
+    def conditionally_print_explanation(self, file_id):
         if self.should_explain_scoring:
             console.print(self.explanation_table, justify="center")
 
+        self.explanation_to_csv(file_id)
+        self.explanation_table_data_for_csv = []
```

Comment:
We can also populate `explanation_table_data_for_csv` directly by looping on `self.explanation_table.rows` if that's available. That can make the code even cleaner